src/pages/tools.js

```diff
@@ -19,18 +18,19 @@ import NavigationConstants from "common/NavigationConstants";
 export default function Tools() {
     const router = useRouter();
     const query = router.query;
-    const { t } = useTranslation("tools");
     const selectedPage = parseInt(query.selectedPage) || 0;
     const selectedRowsPerPage =
         parseInt(getLocalStorage(constants.LOCAL_STORAGE.TOOLS.PAGE_SIZE)) ||
         100;
     const selectedOrder = query.selectedOrder || constants.SORT_ASCENDING;
     const selectedOrderBy = query.selectedOrderBy || "name";
+    const selectedPermFilter = query.selectedPermFilter;
+    const selectedSearchTerm = query.selectedSearchTerm;
 
     const onRouteToListing = useCallback(
-        (order, orderBy, page, rowsPerPage) => {
+        (order, orderBy, page, rowsPerPage, permFilter, searchTerm) => {
             router.push(
-                `/${NavigationConstants.TOOLS}?selectedOrder=${order}&selectedOrderBy=${orderBy}&selectedPage=${page}&selectedRowsPerPage=${rowsPerPage}`
+                `/${NavigationConstants.TOOLS}?selectedOrder=${order}&selectedOrderBy=${orderBy}&selectedPage=${page}&selectedRowsPerPage=${rowsPerPage}&selectedPermFilter=${permFilter}&selectedSearchTerm=${searchTerm}`
```

Comment:
```suggestion
                {
                    pathname: `/${NavigationConstants.TOOLS}`,
                    query: {
                        selectedOrder: order,
                        selectedOrderBy: orderBy,
                        selectedPage: page,
                        selectedRowsPerPage: rowsPerPage,
                        selectedPermFilter: permFilter,
                        selectedSearchTerm: searchTerm,
                    },
                }
```
I also wonder if `pathname: NavigationConstants.TOOLS` would work too.

Reply:
Path needs to start with `/` as far as I know.